compute_sdk/globus_compute_sdk/sdk/executor.py

```diff
@@ -51,29 +53,44 @@ def __executor_atexit():
 threading.Thread(target=__executor_atexit).start()
 
 
-class TaskSubmissionInfo:
+UUID_LIKE_T = t.Union[uuid.UUID, str]
+
+
+def _as_uuid(uuid_like: UUID_LIKE_T) -> uuid.UUID:
+    return uuid_like if isinstance(uuid_like, uuid.UUID) else uuid.UUID(uuid_like)
```

Comment:
The use of `_as_*` in most contexts to which I'm privy suggests a _new_ object.  In this case, if it's already a `uuid.UUID`, then the same object gets returned.  We aren't changing the value of the constructed `UUID` object after the fact, so it's of not consequence here, but you might also consider:
```python
return uuid.UUID(uuid_like)
```
Just run it through the `uuid.UUID` mill and forgo the isinstance check.

Reply:
Unfortunately the `UUID` constructor is not nice enough for that to work:

```
Python 3.10.10 (main, Mar 21 2023, 13:41:39) [Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import uuid
>>> u = uuid.uuid4()
>>> uuid.UUID(u)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/chris/opt/anaconda3/envs/compute/lib/python3.10/uuid.py", line 174, in __init__
    hex = hex.replace('urn:', '').replace('uuid:', '')
AttributeError: 'UUID' object has no attribute 'replace'
>>> 
```

Reply:
Re

> The use of `_as_*` in most contexts to which I'm privy suggests a new object

To my mind, "as" implies only doing a conversion if necessary - see [C#'s `as` keyword](https://stackoverflow.com/q/3160127/5931898). If there's a more Pythonic phrasing, please do let me know.

Reply:
Cool; empirical evidence and different experience counter the suggestion.  Fine with me.